testsuite/MDAnalysisTests/analysis/test_lineardensity.py

```diff
@@ -22,18 +22,91 @@
+                                  15.9994, 1.008, 1.008, 15.9994, 1.008, 1.008,
+                                  15.9994, 1.008, 1.008])
+expected_charges_atoms = np.array([-0.834, 0.417, 0.417, -0.834, 0.417,
+                                   0.417, -0.834, 0.417, 0.417, -0.834,
+                                   0.417, 0.417, -0.834, 0.417, 0.417])
+expected_xpos_atoms = np.array([0., 0., 0., 0.0072334, 0.00473299, 0.,
+                                0., 0., 0., 0.])
+expected_xchar_atoms = np.array([0., 0., 0., 2.2158751e-05, -2.2158751e-05,
+                                 0., 0., 0., 0., 0.])
+
+# test data for grouping='residues'
+expected_masses_residues = np.array([18.0154, 18.0154, 18.0154, 18.0154,
+                                     18.0154])
+expected_charges_residues = np.array([0, 0, 0, 0, 0])
+expected_xpos_residues = np.array([0., 0., 0., 0.00717983, 0.00478656,
+                                   0., 0., 0., 0., 0.])
+expected_xchar_residues = np.array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])
+
+# test data for grouping='segments'
+expected_masses_segments = np.array([90.0770])
+expected_charges_segments = np.array([0])
+expected_xpos_segments = np.array([0., 0., 0., 0.01196639, 0.,
+                                   0., 0., 0., 0., 0.])
+expected_xchar_segments = np.array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])
+
+# test data for grouping='fragments'
+expected_masses_fragments = np.array([18.0154, 18.0154, 18.0154, 18.0154,
+                                      18.0154])
+expected_charges_fragments = np.array([0, 0, 0, 0, 0])
+expected_xpos_fragments = np.array([0., 0., 0., 0.00717983, 0.00478656,
+                                   0., 0., 0., 0., 0.])
+expected_xchar_fragments = np.array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])
+
+
+@pytest.mark.parametrize("grouping, expected_masses, expected_charges,\
+                         expected_xpos, expected_xchar", [
+                         ("atoms",
+                          expected_masses_atoms,
+                          expected_charges_atoms,
+                          expected_xpos_atoms,
+                          expected_xchar_atoms),
+                         ("residues",
+                          expected_masses_residues,
+                          expected_charges_residues,
+                          expected_xpos_residues,
+                          expected_xchar_residues),
+                         ("segments",
+                          expected_masses_segments,
+                          expected_charges_segments,
+                          expected_xpos_segments,
+                          expected_xchar_segments),
+                         ("fragments",
+                          expected_masses_fragments,
+                          expected_charges_fragments,
+                          expected_xpos_fragments,
+                          expected_xchar_fragments)
+                         ])
```

Comment:
PEP8 is a suggestion not an obligation - don't use this against me

```suggestion
@pytest.mark.parametrize("grouping, expected_masses, expected_charges, expected_xpos, expected_xchar", [
    ("atoms", expected_masses_atoms, expected_charges_atoms,
     expected_xpos_atoms, expected_xchar_atoms),
    ("residues", expected_masses_residues, expected_charges_residues,
     expected_xpos_residues, expected_xchar_residues),
    ("segments", expected_masses_segments, expected_charges_segments,
     expected_xpos_segments, expected_xchar_segments),
    ("fragments", expected_masses_fragments, expected_charges_fragments,
     expected_xpos_fragments, expected_xchar_fragments)
])
```